avm/src/vm/memory.rs

```diff
@@ -136,67 +144,120 @@ impl HandlerMemory {
     }
   }
 
-  /// Takes a given address and looks up the `mems` indexes relevant to it.
-  fn addr_to_idxs(self: &HandlerMemory, addr: i64) -> (usize, usize) {
+  pub fn drop_parent(self: &mut HandlerMemory) {
+    if self.parent.is_some() {
+      let arc = self.parent.take();
+      drop(arc);
+    }
+  }
```

Comment:
What is this for? Wouldn't the parent be the one to consume the child?

Reply:
Consuming the children is a mutable operation on the parent that cannot be performed on an object within an Arc. As a result, I have all the forked children drop their reference to the parent, take the parent out of the Arc and then join on its children.